mino/minogrpc/server.go

```diff
@@ -702,16 +678,23 @@ func (mgr *connManager) Acquire(to mino.Address) (grpc.ClientConnInterface, erro
 		grpc.WithStreamInterceptor(
 			otgrpc.OpenTracingStreamClientInterceptor(tracer, otgrpc.SpanDecorator(decorateClientTrace)),
 		),
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
 	}
 
-	if mgr.useTLS {
+	if mgr.serveTLS {
 		ta, err := mgr.getTransportCredential(to)
 		if err != nil {
 			return nil, xerrors.Errorf("failed to retrieve transport credential: %v", err)
 		}
 
 		opts = append(opts, grpc.WithTransportCredentials(ta))
+	} else {
+		// If the remote is accessible via port 443, we suppose it is TLS terminated and signed by a
+		// CA available on the system.
```

Comment:
this assumption might easily break in our own deployments.
that was the whole point of passing the certificate hash along with the token when joining nodes 😕